modules/core/src/smithy4s/internals/DocumentDecoderSchemaVisitor.scala

```diff
@@ -323,6 +322,7 @@ class DocumentDecoderSchemaVisitor(
           val path = PayloadPath.Segment(jLabel) :: pp
           fields
             .get(jLabel) match {
+            case Some(DNull) => buffer(None)
```

Comment:
**issue:** this is exactly what I was thinking, how dare you steal my ideas :P `/s`

Reply:
Haha, great minds think alike? 👀